nisystemlink/clients/core/_uplink/_base_client.py

```diff
@@ -41,7 +42,7 @@ def _handle_http_status(response: Response) -> Optional[Response]:
 @dumps.to_json(JsonModel)
 def _deserialize_model(model_cls: Type[JsonModel], model_instance: JsonModel) -> Dict:
     """Turns a :class:`.JsonModel` instance into a dictionary for serialization."""
-    return model_instance.dict(by_alias=True, exclude_unset=True)
+    return json.loads(model_instance.json(by_alias=True, exclude_unset=True))
```

Comment:
Working around this issue: https://github.com/pydantic/pydantic/issues/1409
tl;dr: calling `dict()` on a model doesn't serialize datetimes like `json()` does, but the converter is expecting a dictionary.